core/src/main/scala/filodb.core/query/RangeVector.scala

```diff
@@ -93,7 +96,92 @@ trait RangeVector {
   def numRows: Option[Int] = None
   def prettyPrint(formatTime: Boolean = true): String = "RV String Not supported"
 }
+trait SerializableRangeVector extends RangeVector {
+  def numRowsInt: Int
+}
+
+trait ScalarVector extends SerializableRangeVector {
+  def key: RangeVectorKey = CustomRangeVectorKey(Map.empty)
+  def getValue(time: Long): Double
+}
+
+case class ScalarVaryingDouble(private val timeValueMap: Map[Long, Double]) extends ScalarVector {
+  override def rows: Iterator[RowReader] = timeValueMap.toList.sortWith(_._1 < _._1).
+    map{ x=> new TransientRow(x._1, x._2)}.iterator
+  def getValue(time: Long): Double = timeValueMap(time)
+
+  override def numRowsInt: Int = timeValueMap.size
+}
 
+final case class RangeParams(start: Long, step: Long, end: Long)
+trait ScalarSingleValue extends ScalarVector {
+  def rangeParams: RangeParams
+  var numRowsInt : Int = 0
+
+  override def rows: Iterator[RowReader] = {
+    val rowList = new ListBuffer[TransientRow]()
```

Comment:
This pattern uses up a whole bunch of memory, which is really not necessary.  It would be far more efficient to do something like the following:

    Iterator.from(rangeParams.start, rangeParams.step).takeWhile(_ <= rangeParams.end).map { i =>
      numRowsInt += 1
      new TransientRow(I*1000, getValue(I*1000))
    }

Reply:
Didn't know we can do this! Thanks!